spring-integration-file/src/test/java/org/springframework/integration/file/remote/session/ContextHolderRequestHandlerAdviceTests.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.file.remote.session;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import org.springframework.integration.support.MessageBuilder;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+public class ContextHolderRequestHandlerAdviceTests {
```

Comment:
This test class does not bring any value.
Or probably better to say: it is just redundant and can be covered by other more realistic tests where we really apply this new advice on endpoints.
See docs for more info: https://docs.spring.io/spring-integration/docs/current/reference/html/messaging-endpoints.html#message-handler-advice-chain

Reply:
I replaced the tests with a new one based on what I understood from the documentation. But I cannot get it to work. Although I added the advice to a service activator, created a delegatingSessionFactory and set the functions in the new advice as shown in the example. But, I am still getting the following error: 

```
Wanted but not invoked:
session.list("foo/");
-> at org.springframework.integration.file.remote.session.ContextHolderRequestHandlerAdviceTests.testDelegates(ContextHolderRequestHandlerAdviceTests.java:73)
Actually, there were zero interactions with this mock.
```

I know the test still does not look good, many parts are unused (and maybe unneeded).

Reply:
So, you have agreed with me about this test class problem, but you did nothing to fix it.

Are you still going?